pandas-stubs/core/dtypes/generic.pyi

```diff
@@ -0,0 +1,15 @@
+from pandas import (
+    DataFrame,
+    Index,
+    Series,
+)
+from pandas.core.arrays import ExtensionArray
+from pandas.core.generic import NDFrame
+
+ABCIndex = type[Index]
+
+ABCNDFrame = type[NDFrame]
+ABCSeries = type[Series]
+ABCDataFrame = type[DataFrame]
+
+ABCExtensionArray = type[ExtensionArray]
```

Comment:
Technically need only `ABCExtensionArray` and `ABCSeries` for the current stubs.